pyscitt/pyscitt/receipt.py

```diff
@@ -21,27 +24,82 @@
 
 HEADER_PARAM_TREE_ALGORITHM = "tree_alg"
 TREE_ALGORITHM_CCF = "CCF"
-
-
-def hdr_as_dict(phdr: dict) -> dict:
+COMMON_CWT_KEYS_MAP = {
+    1: "iss",
+    2: "sub",
+    3: "aud",
+    4: "exp",
+    5: "nbf",
+    6: "iat",
+    7: "cti",
+}
+
+
+def display_cwt_key(item: Any) -> Union[int, str]:
+    """Convert a CWT key to a string for pretty-printing."""
+    out = str(item)
+    return COMMON_CWT_KEYS_MAP.get(item, out)
+
+
+def display_cbor_val(item: Any) -> str:
+    """Convert a CBOR item to a string for pretty-printing."""
+    out = str(item)
+    if hasattr(item, "__name__"):
+        out = item.__name__
+    elif isinstance(item, datetime.datetime):
+        out = item.isoformat()
+    elif type(item) is bytes:
+        out = item.hex()
+    return out
+
+
+def cbor_as_dict(cbor_obj: Any, cbor_obj_key: Any = None) -> Any:
     """
-    Return a representation of a list of COSE header parameters that
-    is amenable to pretty-printing.
+    Return a printable representation of a CBOR object.
     """
 
-    def display(item):
-        if hasattr(item, "__name__"):
-            return item.__name__
-        if type(item) is bytes:
-            return item.hex()
-        return item
+    # pycose will use class instances for known and registered headers instead of ints
+    if hasattr(cbor_obj_key, "identifier"):
+        if cbor_obj_key.identifier == crypto.SCITTReceipts.identifier:
+            parsed_receipts = []
+            for item in cbor_obj:
+                if type(item) is bytes:
+                    receipt_as_dict = Receipt.decode(item).as_dict()
+                else:
+                    receipt_as_dict = Receipt.from_cose_obj(item).as_dict()
+            parsed_receipts.append(receipt_as_dict)
```

Comment:
Shouldn't this statement be included inside the for loop? I think this is just adding the last converted item to the list.

Reply:
Thanks! Great observation, the tests could not detect it as we do not have such examples in code, there is not expectation to use signatures with multiple receipts any time soon